File: cloud-storage-spring-java/src/main/java/com/github/sevtech/cloud/storage/spring/config/ConditionalOnCloudStorageProperty.java
package com.github.sevtech.cloud.storage.spring.config;

import org.springframework.context.annotation.Conditional;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target({ElementType.TYPE, ElementType.METHOD})
@Retention(RetentionPolicy.RUNTIME)
@Conditional(OnPropertyCondition.class)
@interface ConditionalOnCloudStorageProperty {
    String value();

    boolean on() default true;
}

File: cloud-storage-spring-kotlin/src/main/kotlin/com/github/sevtech/cloud/storage/spring/config/ConditionalOnCloudStorageProperty.kt
package com.github.sevtech.cloud.storage.spring.config

import org.springframework.context.annotation.Conditional

@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS, AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@Conditional(OnPropertyCondition::class)
internal annotation class ConditionalOnCloudStorageProperty(val value: String, val on: Boolean = true)